lib/livebook/zta/cloudflare.ex

```diff
@@ -0,0 +1,95 @@
+defmodule Livebook.ZTA.Cloudflare do
+  @moduledoc false
+
+  use GenServer
+  require Logger
+  import Plug.Conn
+
+  @assertion "cf-access-jwt-assertion"
+  @renew_afer 24 * 60 * 60 * 1000
+
+  defstruct [:name, :req_options, :identity]
+
+  def start_link(opts) do
+    identity = identity(opts[:identity][:key])
+    options = [req_options: [url: identity.certs], identity: identity]
+    GenServer.start_link(__MODULE__, options, name: opts[:name])
+  end
+
+  def authenticate(name, conn) do
+    token = get_req_header(conn, @assertion)
+    GenServer.call(name, {:authenticate, token})
```

Comment:
Our GenServer stores things in ets, but we never access them in this function, in the caller. Thus I believe ets is an overkill and we might as well store things in server state. To take full advantage of ets and avoid a single genserver bottleneck, I believe we want something like this:

```elixir
def authenticate(name, conn) do
  token = Plug.Conn.get_req_header(conn, @assertion)
  do_authenticate(name, token) || GenServer.call(name, {:authenticate, token})
end
```

where both `do_authenticate` and `handle_call({:authenticate` fetches things from ets.

Reply:
Good find, added to #1941. Using ets also means the table has to be named, which means we need to require the `:name` option.